### PR TITLE
[Search] Customize <SearchResultGroup/> no results state

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

The `<SearchResultGroup />` component now accepts an optional property `disableRenderingWithNoResults` to disable rendering when no results are returned. Possibility to provide a custom no results component if needed through the `noResultsComponent` property.

Examples:

_Rendering a custom no results component_

```jsx
<SearchResultGroup
  query={query}
  icon={<DocsIcon />}
  title="Documentation"
  noResultsComponent={<ListItemText primary="No results were found" />}
/>
```

_Disable rendering when there are no results_

```jsx
<SearchResultGroup
  query={query}
  icon={<DocsIcon />}
  title="Documentation"
  disableRenderingWithNoResults
/>
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-react/api-report.md [hunk 1]
```diff
@@ -269,13 +271,22 @@ export type SearchResultGroupLayoutProps<FilterOption> = ListProps & {
...
+    array: FilterOption[],
+  ) => JSX.Element | null;
   filterFields?: string[];
   renderFilterField?: (key: string) => JSX.Element | null;
   resultItems?: SearchResult_2[];
-  renderResultItem?: (resultItem: SearchResult_2) => JSX.Element;
+  renderResultItem?: (
+    value: SearchResult_2,
+    index: number,
+    array: SearchResult_2[],
```
Comment (iamEAP) [inline plugins/search-react/api-report.md:285]: It looks like this (and the `renderResultItem` equivalent above) could have been an unintentional API addition as a result of passing the `renderResultItem` handler directly to a `.map`?  ...But perhaps having the `index` and the full array could be useful.

Just want to confirm: do you want to add these two arguments to the API of the `renderResultItem` (and `renderFilterOption`) handler(s)?

**Review by iamEAP — APPROVED** [review]
Cool!  One (very minor) question from me.

File: plugins/search-react/api-report.md (hunk 1 above)
Comment (iamEAP) [reply plugins/search-react/api-report.md:285]: Aaaah, makes sense.  That's great.

**Comment by suuus** [comment]
Thank you for contributing during Hacktoberfest 🍂! You can claim the Backstage Hacktoberfest Holopin at https://bck.st/hacktoberfest-holopin 🙌🏻

**Pull request merged**